scripts/ai_generation/converter.py

```diff
@@ -0,0 +1,461 @@
+
+INPUT: {user's description of a scenario for which they want to leverage Azure} 
+
+OUTPUT: {a list of existing relevant Azure docs (ONLY THAT USE CLI) that, when followed in sequence, will fulfill this scenario COMPLETELY.}
+
+TAKE A DEEP BREATH AND GO!"""
+
+system_prompt_generate_faq = """You are an Azure documentation expert who is great at generating FAQs. Adhere to the following format:
+
+INPUT: {An Azure Document in markdown format}
+
+OUTPUT IN DICT FORMAT: {a list of frequently asked questions, including the most common errors and associated fixes that people would encounter when running the commands given in INPUT, along with specific answers to those questions. every answer should be specific, tangible, and detailed. it may or may not leverage existing resources. include relevant link(s), wherever necessary, in every answer to get more information. OUTPUT SHOULD ONLY HAVE QUESTIONS AND ANSWERS AND OUTPUT SHOULD STRICTLY BE IN DICT FORMAT, WITH KEY BEING THE QUESTION AND VALUE BEING THE ANSWER}
+
+TAKE A DEEP BREATH AND GO!"""
+
+def extract_raw_azure_doc_url(url):
+    response = requests.get(url)
+    soup = BeautifulSoup(response.text, 'html.parser')
+    links = soup.find_all('a')
+    for link in links:
+        github_url = link.get('href') 
+        if github_url is not None and 'github' in github_url and 'md' in github_url:
+            github_url = github_url.replace('blob/', '').replace('github.com', 'raw.githubusercontent.com')
+            return(github_url)
+    
+def get_ai_generated_exec_doc(text=None):
+    response = openai.ChatCompletion.create(
+        engine="gpt-35-turbo-16k",
+        model="gpt-3.5-turbo",
+        messages=[
+            {"role": "system", "content": system_prompt_generate_exec_doc},
+            {"role": "user", "content": f"INPUT: {text}\nOUTPUT DOC:"}
+    ])
+    expected_result = f"Results:\n\n<!-- expected_similarity=0.3 -->\n```json\n{response.choices[0].message.content}\n```" #currently hardcoding the expected similarity to be 0.3 and can change later
+    return expected_result
+
+def get_azure_docs_by_keyword(keyword):
+    repo = g.get_repo('MicrosoftDocs/azure-docs')
+    contents = repo.get_contents('')
+    relevant_azure_docs = {}
+    try:
+        while contents:
+            file_content = contents.pop(0)
+            if file_content.type == "dir":
+                contents.extend(repo.get_contents(file_content.path))
+            else:
+                if keyword.lower() in file_content.download_url and 'github' in file_content.download_url and 'doc' in file_content.download_url and 'md' in file_content.download_url:
+                    if file_content.name not in relevant_azure_docs:
+                        relevant_azure_docs[file_content.name] = file_content.download_url
+                   
+    except KeyboardInterrupt:
+        print(relevant_azure_docs)
```

Comment:
If this code exits early from a keyboard interrupt it will still return `relevant_azure_docs`. Is this expected behavior?

Reply:
will fix that, thanks! that was for testing i believe, need to change it